x-pack/plugins/maps/public/classes/layers/vector_layer/mvt_vector_layer/mvt_vector_layer.tsx

```diff
@@ -426,6 +426,11 @@ export class MvtVectorLayer extends AbstractVectorLayer {
     this._setMbPointsProperties(mbMap, sourceData.tileSourceLayer);
     this._setMbLinePolygonProperties(mbMap, sourceData.tileSourceLayer);
     this._syncTooManyFeaturesProperties(mbMap);
+    const myLayers = mbMap.getStyle().layers.filter((mbLayer:any) => {
+      return this.ownsMbLayerId(mbLayer.id)
+    });
+    const layerIds = myLayers.map(layer=>layer.id)
+    this.getSource().syncSourceStyle(mbMap,layerIds)
```

Comment:
How about moving layerIds into a getter so they are only gathered when needed, instead of on each execution. Also, lets make `syncSourceStyle` optional so its not required.

```
this.getSource().syncSourceStyle?.(mbMap, () => (
  mbMap.getStyle().layers
    .filter((mbLayer:any) => this.ownsMbLayerId(mbLayer.id))
    .map(layer => layer.id)
))
```